service/storage_service.cc

```diff
@@ -2234,6 +2267,59 @@ class topology_coordinator {
         }
     };
 
+    std::variant<join_node_response_params::accepted, join_node_response_params::rejected>
+    validate_joining_node(const node_to_work_on& node) {
+        if (node.rs->state == node_state::replacing) {
+            auto replaced_id = std::get<replace_param>(node.req_param.value()).replaced_id;
+            if (!_topo_sm._topology.normal_nodes.contains(replaced_id)) {
+                return join_node_response_params::rejected {
+                    .reason = ::format("Cannot replace node {} because it is not in the 'normal' state", replaced_id),
+                };
+            }
+        }
+
+        std::vector<sstring> unsupported_features;
+        const auto& supported_features = node.rs->supported_features;
+        std::ranges::set_difference(node.topology->enabled_features, supported_features, std::back_inserter(unsupported_features));
+        if (!unsupported_features.empty()) {
+            slogger.info("raft topology: node {} does not understand some features: {}", node.id, unsupported_features);
```

Comment:
Maybe even `warn` would be appropriate, the admin is trying to do something fishy

Reply:
Done, I also added more `warn`-level logs for other situations where the node is rejected